File: backend/journal-service/internal/modules/journal/domain/entry.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

type Entry struct {
	ID         uuid.UUID   `json:"id"`
	AuthorID   uuid.UUID   `json:"author_id"`
	TripID     *uuid.UUID  `json:"trip_id"`
	PlaceID    *uuid.UUID  `json:"place_id"`
	Latitude   *float64    `json:"latitude"`
	Longitude  *float64    `json:"longitude"`
	Title      *string     `json:"title"`
	Body       string      `json:"body"`
	Mood       *string     `json:"mood"`
	Rating     *int        `json:"rating"`
	OccurredAt time.Time   `json:"occurred_at"`
	MediaIDs   []uuid.UUID `json:"media_ids"`
	Tags       []string    `json:"tags"`
	CreatedAt  time.Time   `json:"created_at"`
	UpdatedAt  time.Time   `json:"updated_at"`
	DeletedAt  *time.Time  `json:"deleted_at,omitempty"`
}

type Reaction struct {
	EntryID   uuid.UUID `json:"entry_id"`
	AuthorID  uuid.UUID `json:"author_id"`
	Emoji     string    `json:"emoji"`
	CreatedAt time.Time `json:"created_at"`
}

type ReactionCount struct {
	Emoji string `json:"emoji"`
	Count int    `json:"count"`
}

File: landmark_app/lib/features/journal/domain/entities/entry.dart
class JournalEntry {
  final String id;
  final String authorId;
  final String? tripId;
  final String? title;
  final String body;
  final String? mood;
  final int? rating;
  final DateTime occurredAt;
  final DateTime createdAt;
  final List<String> tags;

  const JournalEntry({
    required this.id,
    required this.authorId,
    this.tripId,
    this.title,
    required this.body,
    this.mood,
    this.rating,
    required this.occurredAt,
    required this.createdAt,
    this.tags = const [],
  });

  factory JournalEntry.fromJson(Map<String, dynamic> j) => JournalEntry(
    id: j['id'] as String,
    authorId: j['author_id'] as String,
    tripId: j['trip_id'] as String?,
    title: j['title'] as String?,
    body: j['body'] as String? ?? '',
    mood: j['mood'] as String?,
    rating: j['rating'] as int?,
    occurredAt: DateTime.parse(j['occurred_at'] as String),
    createdAt: DateTime.parse(j['created_at'] as String),
    tags: (j['tags'] as List<dynamic>?)?.cast<String>() ?? [],
  );
}
